Lib/unittest/mock.py

```diff
@@ -1062,6 +1062,10 @@ def _calls_repr(self, prefix="Calls"):
         return f"\n{prefix}: {safe_repr(self.mock_calls)}."
 
 
+# Denylist for forbidden attribute names in safe mode
+ATTRIB_DENY_LIST = {name.removeprefix("assert_") for name in dir(NonCallableMock) if name.startswith("assert_")}
```

Comment:
Sorry for being late to the party, but anyways :)

I think that `ATTRIB_DENY_LIST` is an implementation detail, so it should be `_ATTRIB_DENY_LIST`. And also, it is not suited for mutation, isn't it? So, it should be a `frozenset` instead :)

Reply:
Since this PR is already merged, I've opened https://github.com/python/cpython/pull/100819 with this proposal. Feel free to close if it should be public and mutable!

Reply:
It is already non-public by not being included in `__all__`, and the mutability is not really an issue.